S2VX.Game/Play/UserInterface/ScoreProcessor.cs

```diff
@@ -0,0 +1,87 @@
+﻿using osu.Framework.Allocation;
+using osu.Framework.Audio;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using S2VX.Game.Story;
+using S2VX.Game.Story.Note;
+using System;
+using System.Globalization;
+
+namespace S2VX.Game.Play.UserInterface {
+    /// <summary>
+    /// Handles the scoring of game notes and game hold notes - changes cursor
+    /// color, plays hit sounds, and updates score amount
+    /// </summary>
+    public class ScoreProcessor : CompositeDrawable {
+        [Resolved]
+        private S2VXStory Story { get; set; }
+
+        [Resolved]
+        private S2VXCursor Cursor { get; set; }
+
+        // Score should be a double type because during a drag, score may add
+        // very small values that are between 0 and 1. If we int cast or round
+        // this drag value, we'll always get 0.
+        public double Score { get; private set; }
+        private TextFlowContainer TxtScore { get; set; }
+        public S2VXSample Hit { get; private set; }
+        public S2VXSample Miss { get; private set; }
+
+        [BackgroundDependencyLoader]
+        private void Load(AudioManager audio) {
+            Hit = new S2VXSample("hit", audio);
+            Miss = new S2VXSample("miss", audio);
+
+            Margin = new MarginPadding {
+                Horizontal = S2VXGameBase.GameWidth / 60,
+            };
+            InternalChildren = new Drawable[] {
+                TxtScore = new TextFlowContainer(s => s.Font = new FontUsage("default", S2VXGameBase.GameWidth / 20)) {
+                    RelativeSizeAxes = Axes.Both,
+                    RelativePositionAxes = Axes.Both,
+                    TextAnchor = Anchor.CentreRight,
+                    Text = Score.ToString(CultureInfo.InvariantCulture)
+                }
+            };
+        }
+
+        public void AddScore(double moreScore) {
+            Score += moreScore;
+            TxtScore.Text = $"{Math.Round(Score)}";
+        }
+
+        public void Reset() {
+            Score = 0;
+            TxtScore.Text = $"{Math.Round(Score)}";
+            Hit.Reset();
+            Miss.Reset();
+        }
+
+        public void Process(double time, GameNote note) {
+            var notes = Story.Notes;
+            var score = Math.Abs(time - note.HitTime);
+
+            if (score > notes.MissThreshold) { // miss
+                AddScore(notes.MissThreshold);
```

Comment:
I was wondering if we'll ever need to know the state/result and thus would it be nice to have an enum for that (miss, early, late, etc.). But I guess if we're not using it now there's no need in adding in dead code. Just a thought for the future.